dbms/src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1199,6 +1200,53 @@ bool ParserQualifiedAsterisk::parseImpl(Pos & pos, ASTPtr & node, Expected & exp
 }
 
 
+bool ParserSubstitution::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    if (pos->type != TokenType::OpeningCurlyBrace)
+        return false;
+
+    String name;
+    String type;
+    ++pos;
+
+    if (pos->type != TokenType::BareWord)
+    {
+        expected.add(pos, "string literal");
+        return false;
+    }
+
+    name = String(pos->begin, pos->end);
+    ++pos;
+
+    if (pos->type != TokenType::Colon)
+    {
+        expected.add(pos, "colon between name and type");
+        return false;
+    }
+
+    ++pos;
+
+    if (pos->type != TokenType::BareWord)
+    {
+        expected.add(pos, "string literal");
```

Comment:
.

Reply:
But types can be complex. Example: `Array(Tuple(UInt32, String))`. Please add a test after fix.